src/NUnitCommon/nunit.agent.core/Drivers/NUnitFrameworkApi2018.cs

```diff
@@ -121,7 +121,11 @@ public string Run(ITestEventListener? listener, string filter)
         {
             CheckLoadWasCalled();
             log.Info("Running {0} - see separate log file", Path.GetFileName(_testAssemblyPath.ShouldNotBeNull()));
-            Action<string>? callback = null;
+#if NETFRAMEWORK
+            Action<string>? callback = listener != null ? listener.OnTestEvent : null;
+#else
+            Action<string>? callback = listener != null ? listener.OnTestEvent : null;
+#endif
```

Comment:
I don't see any difference between the two?
Note that it can be simplified into:
```suggestion
            Action<string>? callback = listener?.OnTestEvent;
```

Reply:
Using `?.` gives "method group cannot be made nullable" error.